packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -318,14 +318,64 @@ class DateTimePicker extends React.Component {
 
   handleTimeChange(event, time) {
     this.timeValue = time;
-    const validDate = DateUtil.isValidDate(this.dateValue, this.state.dateFormat) && this.isDateTimeAcceptable(DateTimeUtils.convertDateTimeStringToMomentObject(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds, this.initialTimeZone));
+    let validDate = DateUtil.isValidDate(this.dateValue, this.state.dateFormat) && this.isDateTimeAcceptable(DateTimeUtils.convertDateTimeStringToMomentObject(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds, this.initialTimeZone));
     const validTime = DateTimeUtils.isValidTime(this.timeValue, this.props.showSeconds);
-    const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+    let previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+
+    if (event.type === 'keydown') {
+      // Keyboard shortcut is pressed, split the time value into hour and minute inputs
+      const timeArray = this.timeValue.split(':');
+      if (event.key === 'n' || event.key === 'N') {
+        // `N` set Date to today
+        previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+        validDate = true;
+      } else if (event.key === '-' || event.key === '_') {
+        // `-` subtract 1 minute`
+        if (!validDate) {
+          // Date is not valid, set date to today making it valid
+          previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+          validDate = true;
+        }
+
+        if (timeArray[0] === '23' && timeArray[1] === '59') {
+          // Time is 23:59 which means previous datetime should be on the wrong day, lower it by 1
+          previousDateTime = previousDateTime.subtract(1, 'd');
+        }
+
+        const changedDateTime = DateTimeUtils.updateTime(previousDateTime, this.timeValue, this.props.showSeconds);
+        const checkHour = previousDateTime.subtract(1, 'minute').hours();
+        previousDateTime.add(1, 'minute');
+        if (changedDateTime.isDST() && checkHour !== changedDateTime.hours()) {
+          // Datetime falls under DST and hour is wrong using moment logic, so rewrite it using moment logic
+          timeArray[0] = (`0${checkHour}`).slice(-2);
+          this.timeValue = timeArray.join(':');
+        }
+      } else if (event.key === '=' || event.key === '+') {
+        // `+` add 1 minute`
+        if (!validDate) {
+          // Date is not valid, set date to today making it valid
+          previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+          validDate = true;
+        }
+
+        if (timeArray[0] === '00' && timeArray[1] === '00') {
+          // Time is 00:00 which means previous datetime should be on the wrong day, increase it by 1
+          previousDateTime = previousDateTime.add(1, 'd');
```

Comment:
We need to check to ensure that by adding 1 day does not go beyond the MAX supported date of 12/31/2100. For example, if the date is 12/31/2022 and the `+` key is pressed, it would update the date to `01/01/2101` which was previously not possible to enter because it exceed the [max date](https://github.com/cerner/terra-framework/blob/terra-date-picker%404.72.0/packages/terra-date-picker/src/DateUtil.js#L535-L536). 

This workflow should also be handled in terra-date-picker with the hot key changes but it currently does not. It would be best if this logic can be handled only in terra-date-picker so that terra-date-time-picker doesn't need to care about it.

Reply:
Fixed in Date-picker here https://github.com/cerner/terra-framework/pull/1520/commits/ea6e6e82248964185d5d81ae2f63f8e50f6603ed#diff-dafaabc64f600b92fbfa865adda17d41223fea5f885e15310c2e88582c58e0dcR502 
along with adding tests for it

Currently still 100% correctly implemented in date-time-picker for when increasing or decreasing the minute causes us to go over. but tests for that have been added. https://github.com/cerner/terra-framework/pull/1520/commits/d798c38e2975d58fd2abecc55bc87db4aed4ac68#diff-8f8488b16a2c0f0e4089aa01407d27d8ba3116e1f11e277b8d3554ad88972292R977